platform/ext/target/nordic_nrf/common/core/target_cfg.c

```diff
@@ -1289,15 +1289,48 @@ static const uint32_t target_peripherals[] = {
 	dppi_channel_configuration();
 
     /* GPIO pin configuration */
-#ifdef NRF_SPU
-
-	nrf_spu_gpio_config_set(NRF_SPU, 0, TFM_PERIPHERAL_GPIO0_PIN_MASK_SECURE, SPU_LOCK_CONF_LOCKED);
+	uint32_t secure_pins[] = {
+		TFM_PERIPHERAL_GPIO0_PIN_MASK_SECURE,
 #ifdef TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE
-	nrf_spu_gpio_config_set(NRF_SPU, 1, TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE, SPU_LOCK_CONF_LOCKED);
+		TFM_PERIPHERAL_GPIO1_PIN_MASK_SECURE,
+#endif
+#ifdef TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE
+		TFM_PERIPHERAL_GPIO2_PIN_MASK_SECURE,
 #endif
+	};
+
+	/* Note that there are two different API's for SPU configuration */
+#if NRF_SPU_HAS_MEMORY
+
+	for(int port = 0; port < ARRAY_SIZE(secure_pins); port++){
```

Comment:
Rename variable port -> pin ? Maybe?

Reply:
hmm, If you have misread then I am sure I can improve somehow.

But here, port is 0 or 1 and represents P0 or P1.

Reply:
hmm. Ok, I see that now

Reply:
It confuses a bit at first look that the array is called xx_pins and then the iteration variable is called port